src/main/java/space/obminyashka/items_exchange/service/AdvertisementService.java

```diff
@@ -17,23 +15,17 @@ public interface AdvertisementService {
 
     /**
      * Find N random advertisements as thumbnails with filters
-     *
-     * @param amount amount of random advertisement
-     * @param excludeAdvertisementId excluded id of advertisement for ignoring
-     * @param subcategoryId exist id of subcategory for filtering
+     * @param advertisementFindThumbnailsRequest an object that contains all parameters to search
      * @return random 12 advertisement
      */
-    List<AdvertisementTitleDto> findRandomNThumbnails(int amount, UUID excludeAdvertisementId, Long subcategoryId);
+    List<AdvertisementTitleDto> findRandomNThumbnails(AdvertisementFindThumbnailsRequest advertisementFindThumbnailsRequest);
 
     /**
      * Find page of advertisements with same subcategory without request advertisement
-     *
-     * @param excludeAdvertisementId excluded id of advertisement for ignoring
-     * @param subcategoryId exist id of subcategory for filtering
-     * @param pageable see {@link Pageable} for more details
+     * @param advertisementFindThumbnailsRequest an object that contains all parameters to search
      * @return page of advertisements
      */
-    Page<AdvertisementTitleDto> findAllThumbnails(UUID excludeAdvertisementId, Long subcategoryId, Pageable pageable);
+    Page<AdvertisementTitleDto> findAllThumbnails(AdvertisementFindThumbnailsRequest advertisementFindThumbnailsRequest);
```

Comment:
the name of the variable can be shortened to **findAdvsRequest** or **findRequest**